source/common/http/conn_manager_impl.cc

```diff
@@ -284,6 +284,18 @@ RequestDecoder& ConnectionManagerImpl::newStream(ResponseEncoder& response_encod
   ActiveStreamPtr new_stream(new ActiveStream(*this, response_encoder.getStream().bufferLimit(),
                                               std::move(downstream_request_account)));
 
+  accumulated_requests_++;
+  if (config_.maxRequestsPerConnection() > 0 &&
+      accumulated_requests_ >= config_.maxRequestsPerConnection()) {
+    if (codec_->protocol() < Protocol::Http2) {
+      new_stream->state_.saw_connection_close_ = true;
+      drain_state_ = DrainState::Draining;
```

Comment:
Sorry one more question. Is this line actually necessary? I don't think so?

/wait-any

Reply:
Sorry, I will remove it. It will only make sense if we call `onDrainTimeout` explicitly.

Reply:
I checked. It requires.

Reply:
Why is it required? I don't see it being set in other cases where we set saw_connection_close_ to true?

Reply:
In `encodeHeaders` https://github.com/envoyproxy/envoy/blob/2f54f365a0279bc25b696be1b212b5e8374f54b3/source/common/http/conn_manager_impl.cc#L1400.

Reply:
So you are just trying to keep it from printing that log line? If so I would a) add a comment and b) while you are here add a log line here about this type of closure.

/wait

Reply:
without it, the response header has no close connection.

Reply:
Sorry I'm not following you, the code block you reference above will fix that, but there will be an erroneous log message. I think you can add a comment about this and also add a log line.

Reply:
Sure, I will do that. Is DrainState::Closing more sense?

Reply:
I don't think it matters but probably yes I would switch it.

Reply:
Done.